src/riak_ql_inverse_distrib_fns.erl

```diff
@@ -0,0 +1,144 @@
+%% -------------------------------------------------------------------
+%%
+%% riak_ql_inverse_distrib_fns: implementation of inverse distribution functions
+%%                              for the query runner
+%%
+%% Copyright (c) 2017 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(riak_ql_inverse_distrib_fns).
+
+
+-export(['PERCENTILE_DISC'/3,
+         'PERCENTILE_CONT'/3,
+         'MEDIAN'/3,
+         'MODE'/3]).
+-export([fn_arity/1,
+         fn_type_signature/2,
+         fn_param_check/2,
+         supported_functions/0]).
+
+-type invdist_function() :: 'PERCENTILE_CONT'
+                          | 'PERCENTILE_DISC'
+                          | 'MEDIAN'
+                          | 'MODE'.
+-export_type([invdist_function/0]).
+
+-include("riak_ql_ddl.hrl").
+
+supported_functions() ->
+    ['PERCENTILE_DISC', 'PERCENTILE_CONT', 'MEDIAN', 'MODE'].
+
+-spec fn_type_signature(invdist_function(), [riak_ql_ddl:external_field_type()]) ->
+                               riak_ql_ddl:external_field_type() |
+                               {error, term()}.
+fn_type_signature('PERCENTILE_DISC', [ColumnType, double])
+  when ColumnType == sint64;
+       ColumnType == double;
+       ColumnType == timestamp -> ColumnType;
+fn_type_signature('PERCENTILE_CONT', [_ColumnType, double])
```

Comment:
Does the compiler complain about `ColumnType` being unused if this binding is not prefixed with an underscore? Seems like it should be fine with it being used only in the guard, and if so, I'd rather not see the underscore used here.